### ISSUE TITLE
GraphQLInputObjectType as GraphQLOutputType

### ISSUE THREAD

**Issue by brimworks** [graphql-java/graphql-java#245]
Currently GraphQLInputObjectType does not implement GraphQLOutputType. Is there any reason an input object type can't be used as an output type? From looking at the code, it seems that the difference between an GraphQLInputObjectType and a GraphQLObjectType are:

* GraphQLObjectType can implement interfaces.
* GraphQLInputObjectType fields can have default values.
* GraphQLObjectType can have arguments.
* GraphQLObjectType can have a dataFetcher.

Basically, a GraphQLInputObjectType simply has more restrictions (no arguments, no data fetcher, no implementing of interfaces).

Why? I'd like the ability to define my "structures" as input types that can be used both as inputs and outputs. Adding this ability doesn't look like a big undertaking, but I'm wondering if I'm missing something as to why this was not done.

Thanks,
-Brian

**Comment by brimworks** [comment]
It looks like this is a feature of the GraphQL spec:

http://graphql.org/learn/schema/

"The fields on an input object type can themselves refer to input object types, but you can't mix input and output types in your schema."

...so I guess that answers it :).
